mPower2/mPower2/StudyBurstScheduleManager.swift

```diff
@@ -194,20 +194,14 @@ class StudyBurstScheduleManager : TaskGroupScheduleManager {
     }
     
     /// Has the user been shown the motivation survey?
-    public internal(set) var hasCompletedMotivationSurvey : Bool {
-        get {
-            guard !UserDefaults.standard.bool(forKey: "hasCompletedMotivationSurvey")
-                else {
-                    return true
-            }
-            let result = self.scheduledActivities.contains {
-                $0.activityIdentifier == self.studyBurst.motivationIdentifier.stringValue && $0.isCompleted
-            }
-            self.hasCompletedMotivationSurvey = result
-            return result
-        }
-        set {
-            UserDefaults.standard.set(newValue, forKey: "hasCompletedMotivationSurvey")
+    public var hasCompletedMotivationSurvey : Bool {
+        do {
+            let reportIdentifier = self.studyBurst.motivationIdentifier.stringValue
+            let report: SBBReportData? = try self.participantManager.getLatestCachedData(forReport: reportIdentifier)
+            let isNil = (report?.localDate == nil) && (report?.dateTime == nil)
```

Comment:
...I didn't think it was possible for a non-nil SBBReportData from BridgeSDK cache to have both localDate and dateTime be nil--can you save a report item without setting one or the other?

Reply:
*sigh* There is some weird issue where it is not honoring the Optional (nullable) protocol method and Swift is insisting that it is non-null. Since my test implementation is in swift, it has to return a non-null object. Mostly this is handled by checking for the clientData to be non-nil, but in this case, I wanted to use the synchronous get from cache directly b/c otherwise the startup race condition results in the incorrect view controller on launch.